lib/wolfe/cleanup.rb

```diff
@@ -50,31 +52,46 @@ def timespan_regex
       #
 
       def cleanup( config )
-        daily_date = Date.today - eval( config['one_per_day_timespan'] )
-        monthly_date = Date.today - eval( config['one_per_month_timespan'] )
-        first_relevant_date = Date.today - 5.years
+        daily_date = Date.today - eval(config['one_per_day_timespan'])
```

Comment:
There's still an `eval` left.

Reply:
Also, it might be worthwhile to reexamine how this config is stored. While it's more convenient to be able to say `15.days` in the config, it also relies on the "magic" of `eval` to work in Ruby. Using something more structured might be worthwhile, along the lines of:

```yaml
# In config:
backup_name:
  # […]
  one_per_day_timespan:
    duration: 15
    unit: days
  one_per_month_timespan:
    duration: 1
    unit: year
```

```ruby
class TimespanFromConfiguration
  # Add any needed units here
  SUPPORTED_TIME_UNITS = %w(hours days months years).freeze

  def initialize(config)
    @duration = Integer(config['duration'])
    @unit = config['unit']
    # Always use plural
    @unit << 's' unless @unit.end_with? 's'

    raise ArgumentError.new("Invalid time unit #{config['unit'].inspect}, expected one of #{SUPPORTED_TIME_UNITS.join(', ')}") unless supported_time_unit?
  end

  def timespan
    # We know that @duration is an Integer, and @unit is one of ActiveSupport's time methods
    # This is also the place where we could do custom calculations if needed
    @duration.send(@unit)
  end

  private

    def supported_time_unit?
      SUPPORTED_TIME_UNITS.include?(@unit)
    end
end
```

Usage:

```ruby
daily_date = Date.today - TimespanFromConfiguration.new(config['one_per_day_timespan']).timespan
```